packages/xo-server/src/api/xostor.mjs

```diff
@@ -0,0 +1,184 @@
+      packages: _package,
+    })
+  }
+}
+
+export function installDependencies({ host }) {
+  return installOrUpdateDependencies.call(this, host)
+}
+installDependencies.description = 'Install XOSTOR dependencies'
+installDependencies.permission = 'admin'
+installDependencies.params = {
+  hostId: { type: 'string' },
+}
+installDependencies.resolve = {
+  host: ['hostId', 'host', 'administrate'],
+}
+
+export function updateDependencies({ host }) {
+  return installOrUpdateDependencies.call(this, host, 'update')
+}
+updateDependencies.description = 'Update XOSTOR dependencies'
+updateDependencies.permission = 'admin'
+updateDependencies.params = {
+  hostId: { type: 'string' },
+}
+updateDependencies.resolve = {
+  host: ['hostId', 'host', 'administrate'],
+}
+
+export async function formatDisks({ disks, force, host, ignoreFileSystems, provisioning }) {
+  const rawDisks = disks.join(',')
+  const xapi = this.getXapi(host)
+
+  const lvmPlugin = (fnName, args) => pluginCall(xapi, host, 'lvm.py', fnName, args)
+  log.info(`Format disks (${rawDisks}) with force: ${force}`, { hostId: host.id })
+
+  if (force) {
+    await destroyVolumeGroup(xapi, host, force)
+  }
+  await lvmPlugin('create_physical_volume', {
+    devices: rawDisks,
+    ignore_existing_filesystems: String(ignoreFileSystems),
+    force: String(force),
+  })
+
+  await lvmPlugin('create_volume_group', {
+    devices: rawDisks,
+    vg_name: VG_NAME,
+  })
+
+  if (provisioning === ENUM_PROVISIONING.Thin) {
+    await lvmPlugin('create_thin_pool', {
+      lv_name: LV_NAME,
+      vg_name: VG_NAME,
+    })
+  }
+}
+formatDisks.description = 'Format disks for a XOSTOR use'
+formatDisks.permission = 'admin'
+formatDisks.params = {
+  disks: { type: 'array', items: { type: 'string' } },
+  force: { type: 'boolean', optional: true, default: false },
+  hostId: { type: 'string' },
+  ignoreFileSystems: { type: 'boolean', optional: true, default: false },
+  provisioning: { enum: PROVISIONING },
+}
+formatDisks.resolve = {
+  host: ['hostId', 'host', 'administrate'],
+}
+
+export async function create({ description, disksByHosts, force, ignoreFileSystems, name, provisioning, replication }) {
+  const hostIds = Object.keys(disksByHosts)
+  const hosts = hostIds.map(hostId => this.getObject(hostId, 'host'))
+
+  if (!hosts.every(host => host.$pool === hosts[0].$pool)) {
+    throw new Error('All hosts must be in the same pool')
+  }
+
+  const boundInstallDependencies = installDependencies.bind(this)
+  await asyncEach(hosts, host => boundInstallDependencies({ host }), { stopOnError: false })
+
+  const boundFormatDisks = formatDisks.bind(this)
+  await asyncEach(
+    hosts,
+    host => boundFormatDisks({ disks: disksByHosts[host.id], host, force, ignoreFileSystems, provisioning }),
+    {
+      stopOnError: false,
+    }
+  )
+
+  const pool = this.getObject(hosts[0].$pool, 'pool')
+  const master = hosts.find(host => host.id === pool.master)
```

Comment:
the xapi call are always going to the master , you can use any of the host xapi to make the SR_create call